COMPILING.md

```diff
@@ -1,85 +1,319 @@
-Instructions for compiling on a Debian-based system. The package names here are
-valid for Ubuntu 12.10 and may or may not work on your system.
+# General Linux Guide
 
-Building instructions, below, always assume you are running them from the
-Cataclysm:DDA source directory.
+To build Cataclysm from source you will need at least a C++ compiler, some basic developer tools, and necessary build dependencies. The exact package names vary greatly from distro to distro, so this part of the guide is intended to give you higher-level understanding of the process.
+
+## Compiler
+
+You have three major choices here: GCC, Clang and MXE.
+
+  * GCC is almost always the default on Linux systems so it's likely you already have it
+  * Clang is usually faster than GCC, so it's worth installing if you plan to keep up with the latest experimentals
+  * MXE is a cross-compiler, so of any importance only if you plan to compile for Windows on your Linux machine (not covered here)
+
+(Note that your distro may have separate packages e.g. `gcc` only includes the C compiler and for C++ you'll need to install `g++`.)
+
+Cataclysm is targeting C++11 standard and that means you'll need a compiler that supports it. You can easily check if your version of `g++` supports C++11 by running:
+
+    $ g++ --std=c++11
+    g++: fatal error: no input files
+    compilation terminated.
+
+If you get a line like:
+
+    g++: error: unrecognized command line option ‘--std=c++11’
+
+This means you'll need a newer version of GCC (`g++`).
+
+The general rule is the newer the compiler the better.
+
+## Tools
+
+Most distros seem to package essential build tools as either a single package (Debian and derivatives have `build-essential`) or a package group (Arch has `base-devel`). You should use the above if available. Otherwise you'll at least need `make` and figure out the missing dependencies as you go.
+
+You should also install `git` and `gettext` (optional if you don't care about localizations).
+
+If you plan on keeping up with experimentals you should also install `ccache`, which  will considerably speed-up partial builds.
+
+## Dependencies
+
+There are some general dependencies, optional dependencies and then specific dependencies for either curses or tiles builds. The exact package names again depend on the distro you're using, and whether your distro packages libraries and their development files separately (e.g. Debian and derivatives).
+
+Rough list based on building on Arch:
+
+  * General: `gcc-libs`, `glibc`, `zlib`, `bzip2`
+  * Optional: `lua51`
+  * Curses: `ncurses`
+  * Tiles: `sdl2`, `sdl2_image`, `sdl2_ttf`, `sdl2_mixer`, `freetype2`
+
+E.g. for curses build on Debian and derivatives you'll also need `libncurses5-dev`.
+
+## Make flags
+
+Given you're building from source you have a number of choices to make:
+
+  * `NATIVE=` - you should use `linux64` if you're on a 64 bit system, `linux32` for 32 bit systems, and leave it out if you're building on Cygwin
```

Comment:
Uh, you should be able to leave it out on Linuxen unless you're explicitly cross-compiling (e.g. 32-to-64 or 64-to-32) -- I certainly don't include any `NATIVE=anything` in the jenkins builds.